File: leetcode_java/001_two_sum.java
//class Dcoder {
class _001 {
	public static int[] twoSum(int[] nums, int target) {
		for (int i = 0; i < nums.length; i++) {
			for (int j = i + 1; j < nums.length; j++) {
				if (nums[j] == target - nums[i]) {
					return new int[] { i, j };
				}
			}
		}
		throw new IllegalArgumentException("No two sum solution");
	}

	public static void main(String[] args) {
		System.out.println("001");
		int[] nums = {2, 7, 11, 15};
		int target = 9;
		int[] result = twoSum(nums, target);
        System.out.println(result[0] + " " + result[1]);
	}
}

File: leetcode_cpp_dcoder/001_two_sum.cpp
#include <iostream>
#include <algorithm> // std::find_if
#include <vector>
#include <string>
using namespace std;
 //Compiler version g++ 6.3.0

class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> r = vector<int>(2);
        for (int i = 0; i < numbers.size(); ++i) {
            for (int j = i + 1; j < numbers.size(); ++j) {
                if (numbers[i] + numbers[j] == target) {
                    r[0] = i + 1;
                    r[1] = j + 1;
                    return r;
                }
            }
            
        }
        
    }
};

//taken from leetcode's debug playground
string integerVectorToString(vector<int> list, int length = -1) {
    if (length == -1) {
        length = list.size();
    }

    if (length == 0) {
        return "[]";
    }

    string result;
    for(int index = 0; index < length; index++) {
        int number = list[index];
        result += to_string(number) + ", ";
    }
    return "[" + result.substr(0, result.length() - 2) + "]";
}

int main()
{
    vector<int> nums = {
        1, 2, 3, 4, 5, 6};
    int target = 6;

    vector<int> ret = Solution().twoSum(nums, target);

    string out = integerVectorToString(ret);
    cout << out << endl;
    system("pause");
}